internal/app/backgroundpublisher.go

```diff
@@ -18,27 +18,40 @@
 package app
 
 import (
+	"fmt"
 	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
 
 	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
 )
 
 type backgroundPublisher struct {
-	output chan<- types.MessageEnvelope
+	topic  string
+	output chan<- interfaces.BackgroundMessage
 }
 
 // Publish provided message through the configured MessageBus output
-func (pub *backgroundPublisher) Publish(payload []byte, correlationID string, contentType string) {
+func (pub *backgroundPublisher) Publish(payload []byte, context interfaces.AppFunctionContext) error {
```

Comment:
I'm just wondering how this is meant to be called. Is the Publish() method expected to be called not from within an app function and instead by an external entity? If so, will the caller need to create a dummy App Function Context for the Publish()?

Reply:
> I'm just wondering how this is meant to be called. Is the Publish() method expected to be called not from within an app function and instead by an external entity? If so, will the caller need to create a dummy App Function Context for the Publish()?

fakhir, yes, there is a new `BuildContext` API on the ApplicationService interface that will create the needed App Function Context for use here. @AlexCuse , any additional info to add?
https://github.com/edgexfoundry/app-functions-sdk-go/blob/master/pkg/interfaces/service.go#L143

Reply:
Also the documentation for this is here. Note that it hasn't been updated for Ireland changes
https://docs.edgexfoundry.org/2.0/microservices/application/AdvancedTopics/#background-publishing

Reply:
Got it, thanks!